cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -474,6 +458,27 @@ struct timestamp_scalar : chrono_scalar<T> {
   static_assert(is_timestamp<T>(), "Unexpected non-timestamp type");
   using chrono_scalar<T>::chrono_scalar;
 
+  timestamp_scalar() = default;
+
+  /**
+   * @brief Construct a new timestamp scalar object from a duration that is
+   * convertible to T::duration
+   *
+   * @param value Duration representing number of ticks since the UNIX epoch or
+   * another duration that is convertible to timestamps duration
+   * @param is_valid Whether the value held by the scalar is valid
+   * @param stream CUDA stream used for device memory operations.
+   * @param mr Device memory resource to use for device memory allocation
+   */
+  template <typename Duration2>
+  timestamp_scalar(Duration2 const& value,
```

Comment:
Shouldn't this have the same behavior as the `timestamp` type and only allow construction from a duration?

Reply:
timestamp can also be constructed from a different duration as long as the source duration is convertible to the target duration - typically from a lower precision timestamp to a higher precision one.

Reply:
That's not what this constructor is doing though. `Duration2` is allowed to be any type that is convertible to `T::duration`.

Reply:
i'd to do this as the python/cython code expects a timestamp scalar to be constructed from an integral type (and they do not have the duration type yet). hence, this was made to accept both a tick count and a duration that can be converted to `T`s duration.

Reply:
Ah, okay, understood.